src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -141,247 +124,105 @@ static void smart_amp_set_params(struct comp_dev *dev,
 	}
 }
 
-static inline int smart_amp_set_config(struct comp_dev *dev, const char *data,
-				       uint32_t data_size)
-{
-	struct smart_amp_data *sad = comp_get_drvdata(dev);
-	struct sof_smart_amp_config *cfg;
-	uint32_t cfg_size;
-
-	cfg = (struct sof_smart_amp_config *)data;
-	cfg_size = data_size;
-
-	if (cfg_size != sizeof(struct sof_smart_amp_config)) {
-		comp_err(dev, "smart_amp_set_config(): invalid config size %u, expect %u",
-			 cfg_size, sizeof(struct sof_smart_amp_config));
-		return -EINVAL;
-	}
-
-	comp_dbg(dev, "smart_amp_set_config(): config size = %u", cfg_size);
-
-	memcpy_s(&sad->config, sizeof(struct sof_smart_amp_config), cfg,
-		 sizeof(struct sof_smart_amp_config));
-
-	return 0;
-}
-
-static inline int smart_amp_get_config(struct comp_dev *dev, char *data,
-				       uint32_t *data_size)
-{
-	struct smart_amp_data *sad = comp_get_drvdata(dev);
-	uint32_t cfg_size;
-
-	cfg_size = sizeof(struct sof_smart_amp_config);
-
-	if (cfg_size > *data_size) {
-		comp_err(dev, "smart_amp_get_config(): wrong config size %d",
-			 *data_size);
-	}
-
-	*data_size = cfg_size;
-
-	return memcpy_s(data, cfg_size, &sad->config, cfg_size);
-}
-
-static int smart_amp_set_large_config(struct comp_dev *dev, uint32_t param_id, bool first_block,
-				      bool last_block, uint32_t data_offset, const char *data)
+static int smart_amp_set_config(struct processing_module *mod, uint32_t config_id,
+				enum module_cfg_fragment_position pos, uint32_t data_offset_size,
+				const uint8_t *fragment, size_t fragment_size, uint8_t *response,
+				size_t response_size)
 {
-	struct smart_amp_data *sad = comp_get_drvdata(dev);
+	struct comp_dev *dev = mod->dev;
+	struct smart_amp_data *sad = module_get_private_data(mod);
 
-	comp_dbg(dev, "smart_amp_set_large_config()");
+	comp_dbg(dev, "smart_amp_set_config()");
 
-	switch (param_id) {
+	switch (config_id) {
 	case SMART_AMP_SET_MODEL:
-		return ipc4_comp_data_blob_set(sad->model_handler,
-					       first_block,
-					       last_block,
-					       data_offset,
-					       data);
+		return comp_data_blob_set(sad->model_handler, pos,
+					 data_offset_size, fragment, fragment_size);
 	case SMART_AMP_SET_CONFIG:
-		return smart_amp_set_config(dev, data, data_offset);
-	default:
-		return -EINVAL;
-	}
-}
-
-static int smart_amp_get_large_config(struct comp_dev *dev, uint32_t param_id, bool first_block,
-				      bool last_block, uint32_t *data_offset, char *data)
-{
-	comp_dbg(dev, "smart_amp_get_large_config()");
-
-	switch (param_id) {
-	case SMART_AMP_GET_CONFIG:
-		return smart_amp_get_config(dev, data, data_offset);
+		if (fragment_size != sizeof(struct sof_smart_amp_config)) {
+			comp_err(dev, "smart_amp_set_config(): invalid config size %u, expect %u",
+				 fragment_size, sizeof(struct sof_smart_amp_config));
+			return -EINVAL;
+		}
+		comp_dbg(dev, "smart_amp_set_config(): config size = %u", fragment_size);
+		memcpy_s(&sad->config, fragment_size, fragment, fragment_size);
```

Comment:
the first size should be the size of the destination, i.e. `memcpy_s(&sad->config, sizeof(sad->config), fragment, fragment_size);`

Reply:
since we treat it as an error if fragment_size !=sizeof(sad->config) , so actually if we execute to here, the fragment_size should equal sizeof(sad->config). But it is okay to keep your format for standardization.